operators/src/processing/temporal_raster_aggregation/mean_aggregation_subquery.rs

```diff
@@ -74,50 +72,48 @@ impl<T> TemporalMeanTileAccu<T> {
             }
 
             GridOrEmpty::Empty(_) => {
-                let mut accu_grid = self.value_grid.clone().into_materialized_grid();
-
-                for ((acc_value, acc_count), new_value) in accu_grid
-                    .data
-                    .iter_mut()
-                    .zip(self.count_grid.data.iter_mut())
-                    .zip(in_tile_grid.data.iter())
-                {
-                    if in_tile_grid.is_no_data(*new_value) {
-                        *acc_count = 0;
-                    } else {
+                // this could stay empty?
+                let accu_grid = self.value_grid.clone().into_materialized_grid();
+
+                let map_fn = |grid_index, _acc_values_option| {
+                    let new_value_option = in_tile_grid
+                        .get_masked_at_grid_index(grid_index)
+                        .expect("Grid Index was invalid before.");
+                    if let Some(new_value) = new_value_option {
                         let ivf: f64 = new_value.as_();
-                        *acc_value = ivf;
-                        *acc_count = 1;
+                        Some((ivf, 1))
+                    } else {
+                        None
                     }
-                }
+                };
 
-                self.value_grid = accu_grid.into();
+                self.value_grid = accu_grid.map_index_elements(map_fn).into(); // TODO: could also do this parallel
             }
 
-            GridOrEmpty::Grid(accu_grid) => {
-                for ((acc_value, acc_count), new_value) in accu_grid
-                    .data
-                    .iter_mut()
-                    .zip(self.count_grid.data.iter_mut())
-                    .zip(in_tile_grid.data.iter())
-                {
-                    if in_tile_grid.is_no_data(*new_value) {
-                        // The input pixel value is nodata
-                        if !self.ignore_no_data {
-                            // once nodata always nodata
-                            *acc_count = 0;
+            GridOrEmpty::Grid(_) => {
+                let accu_grid = self.value_grid.clone().into_materialized_grid(); // TODO do not clone!
```

Comment:
why do you use the tile as accu?

Reply:
i don't the value_grid is the grid in the accu